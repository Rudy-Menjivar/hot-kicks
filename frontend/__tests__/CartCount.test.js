import { render, screen } from '@testing-library/react';
import CartCount from '../components/CartCount';

describe('<CartCount>', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Matches Snapshot', () => {
    const { container } = render(<CartCount count={12} />);
    expect(container).toMatchSnapshot();
  });
  it('Updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={13} />);
    // ? Vanilla JS & fancier custom matcher
    expect(container.textContent).toBe('13');
    // ? Update the props
    rerender(<CartCount count="14" />);
    // ? Tests if duplicated animation is working
    expect(container).toHaveTextContent('1413');
    // ? Built in findBy utility method keeps looking until query times out
    await screen.findByText('14');
    expect(container).toMatchSnapshot();
  });
});
