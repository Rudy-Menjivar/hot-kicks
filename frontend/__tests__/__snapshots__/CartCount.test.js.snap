// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount> Matches Snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1vw3wro-1 fymtr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1vw3wro-0 dSWdGs count"
      >
        12
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1vw3wro-1 fymtr"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1vw3wro-0 dSWdGs count count-enter count-enter-active"
      >
        14
      </div>
      <div
        class="CartCount__Dot-sc-1vw3wro-0 dSWdGs count count-exit count-exit-active"
      >
        13
      </div>
    </div>
  </span>
</div>
`;
